linkerd/http-route/src/grpc/filter/inject_failure.rs

```diff
@@ -0,0 +1,30 @@
+use std::hash::Hash;
+
+pub use crate::http::filter::Distribution;
+
+/// A filter that responds with an HTTP error.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct InjectFailure {
+    pub response: FailureResponse,
+    pub distribution: Distribution,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct FailureResponse {
+    pub code: u16,
+    pub message: std::sync::Arc<str>,
+}
+
+// === impl InjectFailure ===
+
+impl InjectFailure {
+    pub fn apply(&self) -> Option<FailureResponse> {
+        use rand::distributions::Distribution;
+
+        if self.distribution.sample(&mut rand::thread_rng()) {
+            return Some(self.response.clone());
+        }
+
+        None
+    }
+}
```

Comment:
it occurs to me that we _could_ get away with just having a single 
```rust
pub struct InjectFailure<R: Clone> {
    pub response: R,
    pub distribution: Distribution,
}

impl<R: Clone> InjectFailure<R> {
    pub fn apply(&self) -> Option<R> {
        use rand::distributions::Distribution;

        if self.distribution.sample(&mut rand::thread_rng()) {
            return Some(self.response.clone());
        }

        None
    }
}
```
and use that for both HTTP and gRPC. But, this may be kind of a premature genericization, so :woman_shrugging: